psycopg/psycopg/raw_cursor.py

```diff
@@ -0,0 +1,60 @@
+"""
+psycopg raw queries cursors
+"""
+
+# Copyright (C) 2022 The Psycopg Team
+
+from typing import Optional, TYPE_CHECKING, Sequence, Tuple, List
+
+from ._queries import PostgresQuery, QueryPart
+
+from .abc import ConnectionType, Query, Params
+from .rows import Row
+from .cursor import BaseCursor, Cursor
+from ._enums import PyFormat
+from .cursor_async import AsyncCursor
+from typing import Any
+
+if TYPE_CHECKING:
+    from .connection import Connection  # noqa: F401
+    from .connection_async import AsyncConnection  # noqa: F401
+
+
+class RawPostgresQuery(PostgresQuery):
+    @staticmethod
+    def query2pg(
+        query: bytes, encoding: str, vars: Params
+    ) -> Tuple[bytes, Optional[List[PyFormat]], Optional[List[str]], List[QueryPart]]:
+        """
+        Don't convert Python raw query since it's already in a format Postgres understands.
+        """
+        return query, [PyFormat.AUTO] * len(vars), None, []
```

Comment:
I see: if you don't have `vars` here you can't work out the `want_formats` vector.

What you can do is to return None here, and augment `dump()` to set the formats if `None` (or complain if it's not `None`, but they are a different length).

Reply:
Smart! Fixed in pushed commit.